pkg/trace/sampler/prioritysampler.go

```diff
@@ -104,43 +104,61 @@ func (s *PrioritySampler) Sample(trace pb.Trace, root *pb.Span, env string) bool
 	if samplingPriority > 1 {
 		return sampled
 	}
+	// short-circuiting rare P0 trace chunks that the client passed. The sig is already taken in account
+	// in CountWeightedSig below. This chunk will likely be sampled by the ExceptionSampler
+	if samplingPriority == 0 && clientDroppedP0s {
```

Comment:
todo: do this only if the root is in the chunk, otherwise it needs to pass below